### PR TITLE
DeviceTree: Convert 'generated_dts_board.h' into a source file

### PR DESCRIPTION
**Pull request by SebastianBoe**
extract_dts_includes.py has been generating DT output and then
concatenating it with fixup header files to create
'generated_dts_board.h'.

In this patch we instead introduce a source file named
'generated_dts_board.h' and have it \#include the appropriate DT
output and fixup files.

This results in a simpler system because users can now read
'generated_dts_board.h' as C source code to see how fixup files and
generated DT output relate to each other. Whereas before they would
have to either read documentation or python code to gain the same
understanding.

Also, it reduces the scope and complexity of one of our most bloated
python scripts, extract_dts_includes.py.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: include/generated_dts_board.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#ifdef DTS_APP_FIXUP_FILE
+#include DTS_APP_FIXUP_FILE
+#endif
+
+#ifdef DTS_SHIELDS_FIXUP_FILE
+#include DTS_SHIELDS_FIXUP_FILE
+#endif
+
+#endif
```
Comment (hakonfam) [inline include/generated_dts_board.h]: Add comment matching checked define
```
#endif  /* DEVICE_TREE_BOARD_H */
```

File: include/generated_dts_board.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Not a generated file. Feel free to modify.
+ *
+ * Poorly named for legacy compatibility reasons.
+ */
+
+#ifndef DEVICE_TREE_BOARD_H
```
Comment (hakonfam) [inline include/generated_dts_board.h]: Define does not match file name.

**Review by galak — APPROVED** [review]

**Comment by aurel32** [comment]
Note that this commit also broke openthread support in Zephyr as OT is not compiled with the extra -D options.

**Comment by SebastianBoe** [comment]
@pfalcon , @aurel32 : I'm working on fixing the regressions.

**Pull request merged**